indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1024,7 +1034,9 @@ public void runInternal()
       throws ExecutionException, InterruptedException, TimeoutException, JsonProcessingException
   {
     possiblyRegisterListener();
+    stateManager.setStateIfNoSuccessfulRunYet(SeekableStreamSupervisorStateManager.State.CONNECTING_TO_STREAM);
```

Comment:
Hm, I feel like it would be cleaner if the state manger handled the decision of whether to transition to a particular state based on whether a successful run has occurred or not